lib/rex/io/socket_abstraction.rb

```diff
@@ -182,14 +202,18 @@ def monitor_rsock(threadname = 'SocketMonitorRemote')
 
             begin
               close_write if respond_to?('close_write')
-            rescue IOError
+            rescue StandardError
```

Comment:
Forgot to mention, the Meterpreter error raised when calling `close_write` on an invalid handle is *not* derived from `IOError` so I changed this to `StandardError`.